coreax/util.py

```diff
@@ -70,23 +68,8 @@ def apply_negative_precision_threshold(
     :param precision_threshold: Positive threshold we compare against for precision
     :return: ``x``, rounded to 0.0 if it is between ``-precision_threshold`` and 0.0
     """
-    # Validate inputs
-    x = coreax.validation.cast_as_type(x=x, object_name="x", type_caster=float)
-    precision_threshold = coreax.validation.cast_as_type(
-        x=precision_threshold, object_name="precision_threshold", type_caster=float
-    )
-    coreax.validation.validate_in_range(
-        x=precision_threshold,
-        object_name="precision_threshold",
-        strict_inequalities=False,
-        lower_bound=0,
-    )
-
     if precision_threshold < 0.0:
-        raise ValueError(
-            f"precision_threshold must be positive; value {precision_threshold} given."
-        )
-
+        raise ValueError("precision_threshold must not be negative.")
```

Comment:
Is there any reason not to use the f-string like error message from before?

Reply:
Generally, avoiding printing user-inputs has been the approach taken